packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1227,6 +1249,95 @@ class RenderEditable extends RenderBox {
     }
   }
 
+  /// Sets the screen position of the floating cursor and the text position
+  /// closest to the cursor.
+  void setFloatingCursor(TextCursorState state, Offset boundedOffset, TextPosition lastTextPosition) {
+    if (state == TextCursorState.Start) {
```

Comment:
Be liberal with asserts (such as lastTextPosition can't be null, you can't set it to start twice etc etc)